test/test_quantization.py

```diff
@@ -261,6 +261,166 @@ def checkQuantized(model):
         model = quantize(ManualQuantModel().eval(), test_only_eval_fn, self.calib_data)
         checkQuantized(model)
 
+class PostTrainingDynamicQuantTest(QuantizationTestCase):
+    def test_single_layer_dynamic(self):
+        r"""Dynamic Quantize SingleLayerLinearModel which has one Linear module,
+        make sure it is swapped to nnqd.Linear which is the quantized version of
+        the module
+        """
+        model = SingleLayerLinearModel().eval()
+        qconfig_dict = {
+            '': default_qconfig
+        }
+        model = prepare_dynamic(model, qconfig_dict)
+        # Check if observers are inserted
+        self.checkObservers(model)
+
+        test_only_eval_fn(model, self.calib_data)
+        convert_dynamic(model)
+
+        def checkQuantized(model):
+            self.checkDynamicQuantizedLinear(model.fc1)
+            test_only_eval_fn(model, self.calib_data)
+
+        checkQuantized(model)
+
+        # test one line API
+        model = quantize_dynamic(SingleLayerLinearModel().eval(), test_only_eval_fn, self.calib_data, qconfig_dict)
+        checkQuantized(model)
+
+    def test_two_layers(self):
+        r"""TwoLayerLinearModel has two Linear modules but we only quantize the second one
+        `fc2`, and `fc1`is not quantized
+        """
+        model = TwoLayerLinearModel().eval()
+        qconfig_dict = {
+            'fc2': default_qconfig
+        }
+        model = prepare_dynamic(model, qconfig_dict)
+
+        self.checkObservers(model)
+
+        test_only_eval_fn(model, self.calib_data)
+        convert_dynamic(model)
+
+        def checkQuantized(model):
+            self.assertEqual(type(model.fc1), torch.nn.Linear)
+            self.checkDynamicQuantizedLinear(model.fc2)
+            test_only_eval_fn(model, self.calib_data)
+
+        checkQuantized(model)
+
+        # test one line API
+        model = quantize_dynamic(TwoLayerLinearModel().eval(), test_only_eval_fn, self.calib_data, qconfig_dict)
+        checkQuantized(model)
+
+    def test_nested1(self):
+        r"""Test quantization for nested model, top level 'fc3' and
+        'fc1' of submodule 'sub2', 'sub2.fc2' is not quantized
+        """
+        model = NestedModel().eval()
+        qconfig_dict = {
+            'fc3': default_qconfig,
+            'sub2.fc1': default_qconfig
+        }
+
+        def checkPrepModules(model, before_calib=False):
+            if before_calib:
+                self.checkObservers(model)
+
+        model = prepare_dynamic(model, qconfig_dict)
+        checkPrepModules(model, True)
+        test_only_eval_fn(model, self.calib_data)
+        convert_dynamic(model)
+
+        def checkQuantized(model):
+            self.checkLinear(model.sub1.fc)
+            self.checkDynamicQuantizedLinear(model.fc3)
+            self.checkDynamicQuantizedLinear(model.sub2.fc1)
+            self.checkLinear(model.sub2.fc2)
+            test_only_eval_fn(model, self.calib_data)
```

Comment:
Please remove all calls to test_only_eval_fn: We do not need this for dynamic quant.

Reply:
Updated.